apps/passport-client/components/screens/VerifyScreen.tsx

```diff
@@ -100,81 +109,73 @@ export function VerifyScreen() {
-    );
+function VerifiedTicket({ pcd, publicKeyName, category }: { pcd: EdDSATicketPCD, publicKeyName: string, category: SpecialTicketCategory }) {
+  if (category === SpecialTicketCategory.Zuzalu23) {
+    return <ZuzaluVerifiedDetails pcd={pcd} publicKeyName={publicKeyName} />
   }
-
-  const valid = await verify(deserializedPCD);
-  if (!valid) {
-    return { valid: false, type: "identity-proof", message: "Invalid proof" };
+  else if (category === SpecialTicketCategory.Zuconnect23) {
+    return <ZuconnectVerifiedDetails pcd={pcd} publicKeyName={publicKeyName} />;
   }
+  return (
+    <CardContainerExpanded>
+      <CardOutlineExpanded>
+        <CardHeader col="var(--accent-lite)">
+          <div>VERIFIED</div>
+        </CardHeader>
+      </CardOutlineExpanded>
+    </CardContainerExpanded>
+  );
+}
 
-  // Verify identity proof
-  const payload = JSON.parse(deserializedPCD.claim.signedMessage) as QRPayload;
-  const uuid = bigintToUuid(BigInt(payload.uuid));
-  const userResult = await requestUser(appConfig.zupassServer, uuid);
-
-  if (userResult.error?.userMissing) {
-    return {
-      valid: false,
-      type: "identity-proof",
-      message: "User not found"
-    };
-  }
+async function deserializeAndVerify(pcdStr: string, pcds: PCDCollection): Promise<VerifyResult> {
+  console.log(pcdStr);
+  const decodedPCD = decodeQRPayload(pcdStr);
 
-  if (!userResult.success) {
-    console.log("error lodaing user", userResult.error);
-    return {
-      valid: false,
-      type: "identity-proof",
-      message: "Error loading user"
-    };
-  }
+  const result = await requestVerifyTicket(
+    appConfig.zupassServer,
+    {
+      pcd: decodedPCD
+    }
+  );
 
-  if (
-    userResult.value.commitment !== deserializedPCD.claim.identityCommitment
-  ) {
-    return {
-      valid: false,
-      type: "identity-proof",
-      message: "User doesn't match proof"
-    };
+  if (result.success && result.value.verified) {
+    const pcd = await pcds.deserialize(JSON.parse(decodedPCD));
+    if (isEdDSATicketPCD(pcd)) {
+
+      if (result.value.verifiedTicket) {
+        return {
+          outcome: VerifyOutcome.VerifiedTicket,
+          pcd,
+          publicKeyName: result.value.publicKeyName,
+          category: result.value.category
+        }
+      } else {
+        return {
+          outcome: VerifyOutcome.VerifiedPCDOnly,
+          pcd
+        }
+      }
+    }
   }
 
-  const timeDifferenceMs = Date.now() - payload.timestamp;
-
-  if (timeDifferenceMs >= appConfig.maxIdentityProofAgeMs) {
-    return {
-      valid: false,
-      type: "identity-proof",
-      message: "Proof expired"
-    };
+  return {
+    outcome: VerifyOutcome.NotVerified,
+    message: (result.success && result.value.verified === false ? result.value.message : null) ?? "Could not verify PCD"
   }
-
-  return { valid: true, type: "identity-proof", user: userResult.value };
 }
+
+const EventName = styled.span`
+  color: var(--white);
+`;
```

Comment:
make sure you've got the autoformatter running

Reply:
I have whatever VSCode picks up from the config files. I've noticed that it seems to tolerate very long lines sometimes, and other times reformats them, and I don't really know why. I tend to restart VSCode more frequently, perhaps that helps.